app.js

```diff
@@ -34,10 +94,122 @@ app.post('/vegetables', helpers.validate, (req, res, next) => {
   const { vegetables } = data
   const vegetable = { id: generateId(), ...req.body }
 
-  vegetables.push(vegetable)
+  vegetables.push(vegetable)                       
   res.status(201).json(vegetable)
+})                         
+                            
+app.delete('/vegetables/:id', ( req, res, next) => { 
+  const { vegetables } = data
+  const { id } = req.params
+
+  const veg = vegetables.filter(el => el.id.includes(id))[0]
+  if (!veg) {
+    const message = `Could not find vegetable with ID of ${id}`
+    next({ status: 404, message })
+  }
+  const vegIndex = vegetables.indexOf(veg)
+  vegetables.splice(vegIndex,1)
+  res.status(200).json(veg)
 })
 
+app.put('/vegetables/:id', helpers.validate, (req, res, next) => {
```

Comment:
I think `helpers.validate` should perform the same function as the lines you have below.